windows/src/desktop/kmshell/main/OnlineUpdateCheck.pas

```diff
@@ -372,6 +618,26 @@ procedure TOnlineUpdateCheck.DoInstallKeyman;
     ShowMessage(SysErrorMessage(GetLastError));
 end;
 
+function TOnlineUpdateCheck.DoInstallKeyman(SavePath: string) : Boolean;
+var
+  s: string;
+  FResult: Boolean;
+begin
+  s := LowerCase(ExtractFileExt(SavePath));
+  if s = '.msp' then
+    FResult := TUtilExecute.Shell(0, 'msiexec.exe', '', '/qb /p "'+SavePath+'" AUTOLAUNCHPRODUCT=1')  // I3349
```

Comment:
We can delete this because we don't distribute .msp patch files any longer (and never will, it's a fragile technology).